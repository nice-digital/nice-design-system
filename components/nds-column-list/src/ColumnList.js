import React from "react";
import PropTypes from "prop-types";
import "../scss/column-list.scss";

export const ColumnList = props => {
	const { children, plain, className, columns, ...attrs } = props;

	return (
		<ol
			className={[
				"column-list",
				plain ? "column-list--plain" : "column-list--boxed",
				columns === 2 ? "column-list--two-columns" : "",
				className
			]
				.filter(Boolean)
				.join(" ")}
			{...attrs}
		>
			{children}
		</ol>
	);
};

ColumnList.propTypes = {
	children: PropTypes.node.isRequired,
	plain: PropTypes.bool,
	className: PropTypes.string,
	columns: PropTypes.number
};
