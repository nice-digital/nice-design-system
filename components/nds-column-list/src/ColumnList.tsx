import React from "react";
import "../scss/column-list.scss";

export interface ColumnListProps {
	children: React.ReactNode;
	plain?: boolean;
	className?: string;
	/** Number of columns on desktop-width screens */
	columns?: 2 | 3;
	[prop: string]: unknown;
}

export const ColumnList: React.FC<ColumnListProps> = (
	props: ColumnListProps
) => {
	const { children, plain, className, columns, ...attrs } = props;

	return (
		<ol
			className={[
				"column-list",
				plain ? "column-list--plain" : "column-list--boxed",
				columns === 2 ? "column-list--two-columns" : "",
				className
			]
				.filter(Boolean)
				.join(" ")}
			{...attrs}
		>
			{children}
		</ol>
	);
};
