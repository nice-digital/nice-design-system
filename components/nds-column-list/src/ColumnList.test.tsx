import React from "react";
import { render } from "@testing-library/react";
import { ColumnList } from "./ColumnList";

const ListItems = () => (
	<>
		<li>One</li>
		<li>Two</li>
		<li>Three</li>
		<li>Four</li>
		<li>Five</li>
		<li>Six</li>
	</>
);

describe("ColumnList", () => {
	it("should match snapshot", () => {
		const wrapper = render(
			<ColumnList>
				<ListItems />
			</ColumnList>
		);
		expect(wrapper).toMatchSnapshot();
	});

	it("should add CSS module class name to parent list for standard (boxed) variant", () => {
		const wrapper = render(
			<ColumnList>
				<ListItems />
			</ColumnList>
		);
		expect(wrapper.getByRole("list")).toHaveClass("column-list--boxed");
	});

	it("should add CSS module class name to parent list for the plain variant", () => {
		const wrapper = render(
			<ColumnList plain>
				<ListItems />
			</ColumnList>
		);
		expect(wrapper.getByRole("list")).toHaveClass("column-list--plain");
	});

	it("should add CSS module class name to parent list for the 2 column variant", () => {
		const wrapper = render(
			<ColumnList columns={2}>
				<ListItems />
			</ColumnList>
		);
		expect(wrapper.getByRole("list")).toHaveClass("column-list--two-columns");
	});

	it("should add no CSS module class name to parent list for the 3 column variant", () => {
		const wrapper = render(
			<ColumnList columns={3}>
				<ListItems />
			</ColumnList>
		);
		expect(wrapper.getByRole("list")).not.toHaveClass(
			"column-list--two-columns"
		);
	});

	it("should append className prop to rendered class attribute", () => {
		const wrapper = render(
			<ColumnList className="test">
				<ListItems />
			</ColumnList>
		);
		expect(wrapper.getByRole("list")).toHaveClass("test");
	});

	it("should render additional props as attributes", () => {
		const wrapper = render(
			<ColumnList aria-label="hello">
				<ListItems />
			</ColumnList>
		);
		expect(wrapper.getByRole("list").getAttribute("aria-label")).toBe("hello");
	});
});
