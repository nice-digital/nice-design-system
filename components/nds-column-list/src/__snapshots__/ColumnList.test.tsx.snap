// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnList should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ol
        class="column-list column-list--boxed"
        data-component="column-list--undefined"
      >
        <li>
          One
        </li>
        <li>
          Two
        </li>
        <li>
          Three
        </li>
        <li>
          Four
        </li>
        <li>
          Five
        </li>
        <li>
          Six
        </li>
      </ol>
    </div>
  </body>,
  "container": <div>
    <ol
      class="column-list column-list--boxed"
      data-component="column-list--undefined"
    >
      <li>
        One
      </li>
      <li>
        Two
      </li>
      <li>
        Three
      </li>
      <li>
        Four
      </li>
      <li>
        Five
      </li>
      <li>
        Six
      </li>
    </ol>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
